zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -78,6 +83,58 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Represents the identifier for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HDSeedAccount(
+    pub HDSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
+
+/// Represents an arbitrary account for which the seed and ZIP-32 account ID are not known
+/// and may not have been involved in creating this account.
+#[derive(Debug, Clone)]
+pub enum ImportedAccount {
+    /// An account that was imported via its full viewing key.
+    Full(UnifiedFullViewingKey),
+}
+
+/// The inputs for adding an account to a wallet.
+#[derive(Debug, Clone)]
+pub enum Account {
+    /// Inputs for a ZIP-32 HD account.
+    Zip32(HDSeedAccount),
+    /// Inputs for an imported account.
+    Imported(ImportedAccount),
+}
+
+impl Account {
+    /// Gets the default UA for the account.
```

Comment:
(From my previous review of this in #1202.)
```suggestion
    /// Returns the default Unified Address for the account, along with the diversifier
    /// index that generated it.
    ///
    /// The diversifier index may be non-zero if the Unified Address includes a Sapling
    /// receiver, and there was no valid Sapling receiver at diversifier index zero.
```